src/crypto-providers/util.ts

```diff
@@ -423,17 +414,41 @@ const getAddressParameters = (
   address: Buffer,
   network: Network,
 ): _AddressParameters | null => {
+  if (hwSigningData == null || hwSigningData.length === 0) return null
+  const { paymentSigningFiles, stakeSigningFiles } = filterSigningFiles(hwSigningData)
   const addressType = getAddressType(address)
+  const findMatchingAddress = (packedAddresses: (_AddressParameters | null)[]) => packedAddresses.find(
+    (packedAddress) => packedAddress && Buffer.compare(packedAddress.address, address) === 0,
+  ) || null
+
   try {
     switch (addressType) {
       case AddressTypes.BOOTSTRAP:
-        return _packBootStrapAddress(hwSigningData[0], network)
+        return findMatchingAddress(
+          paymentSigningFiles.map((paymentSigningFile) => _packBootStrapAddress(paymentSigningFile, network)),
+        )
+
       case AddressTypes.BASE:
-        return _packBaseAddress(hwSigningData, network)
+        return findMatchingAddress(
+          paymentSigningFiles.flatMap((paymentSigningFile) => (
+            stakeSigningFiles.map((stakeSigningFile) => (
+              _packBaseAddress(paymentSigningFile, stakeSigningFile, network)
+            ))
+          )),
+        )
+
       case AddressTypes.ENTERPRISE:
-        return _packEnterpriseAddress(hwSigningData[0], network)
+        return findMatchingAddress(
+          paymentSigningFiles.map((paymentSigningFile) => (
+            _packEnterpriseAddress(paymentSigningFile, network)
+          )),
+        )
+
       case AddressTypes.REWARD:
-        return _packRewardAddress(hwSigningData, network)
+        return findMatchingAddress(
+          stakeSigningFiles.map((stakeSigningFile) => _packRewardAddress(stakeSigningFile, network)),
+        )
+
       default: return null
```

Comment:
We are not supporting pointer addresses? If it is intentional, it should be mentioned here in a comment, perhaps with a brief explanation why we have decided so.

Reply:
Actually I don't know. I'm not even sure what logic for pointer address would be, so I added TODO for now.